### PR TITLE
Add ruleR3020, R3021

### PR DESCRIPTION
**Pull request by ruowan**

### CODE REVIEW

File: src/typescript/azure-openapi-validator/rules/PathResourceTypeNameCamelCase.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    }
+  }
+});
+
+function resourceTypeMustCamelCase(resourceType: string): boolean {
+  if (resourceType.length === 0) {
+    return true;
+  }
+  const pascalCase: RegExp = new RegExp("^[a-z]+(?:[A-Z][a-z]+)*$");
```
Comment (lirenhe) [inline src/typescript/azure-openapi-validator/rules/PathResourceTypeNameCamelCase.ts]: Is this RegExp correct?

File: src/typescript/azure-openapi-validator/rules/PathResourceProviderNamePascalCase.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            message: `${msg} Path: '${it}'`,
+            location: path.concat(["paths"])
+          };
+        }
+      }
+    }
+  }
+});
+
+function resourceProviderMustPascalCase(resourceProvider: string): boolean {
```
Comment (lirenhe) [inline src/typescript/azure-openapi-validator/rules/PathResourceProviderNamePascalCase.ts]: Could you move this function into a common place?

File: src/typescript/azure-openapi-validator/rules/PathResourceTypeNameCamelCase.ts (hunk 1 above)
Comment (ruowan) [reply src/typescript/azure-openapi-validator/rules/PathResourceTypeNameCamelCase.ts]: Updated regex.

File: src/typescript/azure-openapi-validator/rules/PathResourceProviderNamePascalCase.ts (hunk 2 above)
Comment (ruowan) [reply src/typescript/azure-openapi-validator/rules/PathResourceProviderNamePascalCase.ts]: Good idea. Added some unittest to cover.

File: src/typescript/azure-openapi-validator/rules/PathResourceTypeNameCamelCase.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    if (node.paths !== undefined) {
+      const msg: string = "Resource type naming must follow camel case.";
+      const paths: string[] = Object.keys(node.paths);
+      for (const it of paths) {
+        const allWords = getAllWordsFromPath(it);
+        const resourceProviders = new Set<string>(
+          getAllResourceProvidersFromPath(it)
+        );
+        const resourceTypes = allWords.filter(
+          subPath => !resourceProviders.has(subPath)
```
Comment (jianyexi) [inline src/typescript/azure-openapi-validator/rules/PathResourceTypeNameCamelCase.ts]: why can't we use  resourceProviders directly ?

Comment (ruowan) [reply src/typescript/azure-openapi-validator/rules/PathResourceTypeNameCamelCase.ts]: This rule is about resource type camel case. we need to exclude resource provider from all words set.

File: src/typescript/azure-openapi-validator/rules/utilities/rules-helper.ts [hunk 4]
```diff
@@ -28,15 +30,43 @@ function getMostSuccessfulResponseKey(responses: string[]): string {
...
+  resourceProvider: string
+): boolean {
+  if (resourceProvider.length === 0) {
+    return false;
+  }
+  const pascalCase: RegExp = new RegExp(`^[A-Z][a-z]+(?:\.[A-Z]+[a-z]+)*$`);
+  return pascalCase.test(resourceProvider);
+}
+
+export function resourceTypeMustCamelCase(resourceType: string): boolean {
```
Comment (lirenhe) [inline src/typescript/azure-openapi-validator/rules/utilities/rules-helper.ts]: do we support letter in the name?

Comment (ruowan) [reply src/typescript/azure-openapi-validator/rules/utilities/rules-helper.ts]: Can you provide an example of this case?

**Pull request merged**